src/Components/Autocomplete.tsx

```diff
@@ -0,0 +1,107 @@
+import { useMemo, useRef, useState } from 'react';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+type Props = {
+  onSelected: (user: Person | null) => void;
+  delay: number;
+};
+export const Autocomplete: React.FC<Props> = ({ onSelected, delay }) => {
+  const [list, setList] = useState<Person[]>(peopleFromServer);
+  const [query, setQuery] = useState('');
+  const [applayedQuery, setApplayedQuery] = useState('');
+  const [focused, setFocused] = useState(false);
```

Comment:
Consider renaming to isFocused

https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig